### ISSUE TITLE
New error after updating branch from 6.7.0 to 7.0.1 - listSession throws error after initialization

### ISSUE THREAD

**Issue by gabrielaraujoz** [RodrigoSMarques/flutter_branch_sdk#272]
**Describe the bug**
After updating the library from version 6.7.0 to version 7.0.1 and adding the sdk initialization before listSession, I'm getting this exception on Android devices:

FlutterError - PlatformException(-118, Warning. Session initialization already happened. To force a new session, set intent extra, "branch_force_new_session", to true.,)

I always initialize the sdk **before** listSession, but I'm getting this Platform Exception.

I have found [this reference](https://help.branch.io/developers-hub/docs/android-troubleshooting#handle-the-error---brancherrorerr_branch_already_initialized) for solving it on Android but I'm not sure how to do that on Flutter, maybe the SDK should handle this?

**To Reproduce**
1. Init branch normally in Android app
2. Execute `FlutterBranchSDK.listSession()`
3. Watch your console for the following error:
```
W/System.err( 9124): java.lang.IllegalStateException: Reply already submitted
W/System.err( 9124): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:435)
W/System.err( 9124): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:263)
W/System.err( 9124): 	at br.com.rsmarques.flutter_branch_sdk.MethodResultWrapper$1.run(MethodResultWrapper.java:25)
W/System.err( 9124): 	at android.os.Handler.handleCallback(Handler.java:942)
W/System.err( 9124): 	at android.os.Handler.dispatchMessage(Handler.java:99)
W/System.err( 9124): 	at android.os.Looper.loopOnce(Looper.java:201)
W/System.err( 9124): 	at android.os.Looper.loop(Looper.java:288)
W/System.err( 9124): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
W/System.err( 9124): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 9124): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
W/System.err( 9124): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Expected behavior**
The listSession method should not try throw any error

**Screenshots**
Branch setup:
![Screenshot 2023-12-15 at 10 53 40](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/20102814/c9d75aee-5b36-43b3-a66d-46284fdc55c5)

**Smartphone (Please complete the following information. remove session if not platform):**
 - Flutter: version: 3.13.9
 - Package version: 7.0.1
 - OS: Android API 33
 - Device: Google Pixel
 
 - **Additional context**
Extra logs:
![Screenshot 2023-12-15 at 14 12 18](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/20102814/c5623f2f-e8d2-492b-b463-f8997fcde631)

Initialization attempt on main.dart, resulting on the same behavior:
![Screenshot 2023-12-15 at 14 15 46](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/20102814/7779fcf4-4ead-4ee8-bab8-f859dcd3237d)

**Comment by gabrielaraujoz** [comment]
update here: I found out this is more likely to happen in simulators or using Flutter's hot reload, I'm investigating this further

**Comment by gabrielaraujoz** [comment]
another one: upon testing in release mode, this error happens when app is backgrounded and then foregrounded. Is there any way to correct this?

**Comment by RodrigoSMarques** [comment]
@gabrielaraujoz 

Check in `AndroidManifest.xml` if `android:launchMode="singleTask"`

There is a new version that has not yet been published, as I am waiting for the results of tests from other users.

Can you do a test?


Update `pubspec.yaml`, according to the example below:

```dart
#flutter_branch_sdk:  ^7.0.1
flutter_branch_sdk:
     git:
       url: https://github.com/RodrigoSMarques/flutter_branch_sdk.git
       ref: v7.0.2-dev
```

**Comment by RodrigoSMarques** [comment]
Fixed in version [7.0.2](https://pub.dev/packages/flutter_branch_sdk)
